MonoGame.Framework/Platform/Graphics/Shader/ShaderProgramCache.cs

```diff
@@ -37,7 +37,7 @@ public int GetUniformLocation(string name)
     /// </summary>
     internal class ShaderProgramCache : IDisposable
     {
-        private readonly Dictionary<int, ShaderProgram> _programCache = new Dictionary<int, ShaderProgram>();
+        private readonly Dictionary<(int, int), ShaderProgram> _programCache = new Dictionary<(int, int), ShaderProgram>();
```

Comment:
I don't like the idea of changing this into a Tuple.

I would do what was suggested in the original issue report (https://github.com/MonoGame/MonoGame/issues/8555) and implement a better hash.

Or store a `ulong` and pack the two `int`s into it if we want to avoid a hash here.

Reply:
The original idea that was suggested could still potentially cause a collision.

Let's just go the ulong route and completely eliminate the chance.

Ill supply the update.

Reply:
Okay they update is in 4ad91310c88323a7b4a0602a4c9905518537afd1